File: 1.에라토스테네스의 체/eratos.py
# -*- coding: utf-8 -*-
import sys, time, timeit

def getEratos(num) :
    #init
    sieve = {}
    for i in range(2, N+1):
        sieve[i] = 0
        
    # Sieve of Eratosthenes
    for i in range(2, N+1):
        if sieve[i]==0:
            n=2
            while i*n <= N:
                sieve[i*n]=1
                n+=1

    #print results
    for i in range(2, N+1):
        if sieve[i]==0:
            print( i, end=" " )

if __name__=="__main__":
    print( sys.version )
    N = int( input("input a number : ") )
    #start = time.time()
    start = timeit.default_timer()
    getEratos(N)
    #end = time.time() - start
    end = timeit.default_timer()
    print("\n<elapsed time> : ", round(end - start, 4))
    #input("press any key")

File: 1.에라토스테네스의 체/Eratos.java
import java.util.Scanner;

public class Eratos {
    public static void main(String[] args) {
        int num = 100;

//        if (args.length > 0)
//            num = Integer.parseInt(args[0]);

        System.out.print("input a number : ");
        Scanner scn = new Scanner(System.in);
        num = scn.nextInt();
        //num = System.in.read();

        long start = System.currentTimeMillis();
        getEratos(num);
        long end = System.currentTimeMillis();
        System.out.println( "<elapsed time> : " + ( end - start )/1000.0 ); 
    }

    public static void getEratos(int max) {
        boolean[] a = new boolean[max];

        for (int i = 2; i < max; i++) 
            a[i] = true;
            
        int to = (int)Math.sqrt(max);
        
        for (int i = 2; i < to; i++)
            if (a[i] != false)

        for (int j = i; j*i < max; j++) 
            a[i*j] = false;
                    
        for (int i = 2; i < max; i++)
            if (a[i]) 
                System.out.print(" " + i);

        System.out.println();
    }

}
